Add warning about duplicating paths across locales (#5487)
mintlify/docs: guides/internationalization.mdx

The navigation structure can differ between languages to accommodate language-specific content needs.

<Warning>

>   Do not use the same page path in more than one locale. Duplicating paths across locales results in undefined behavior. Each page path should appear in only one language's navigation.

Reviewer: ```suggestion
  Do not use the same page path in more than one language. Duplicating paths across languages results in undefined behavior. Each page path must appear in only one language's navigation.
```
